prometheus/src/main/java/org/opensearch/sql/prometheus/response/PrometheusResponse.java

```diff
@@ -44,10 +68,15 @@ public Iterator<ExprValue> iterator() {
         for (int j = 0; j < values.length(); j++) {
           LinkedHashMap<String, ExprValue> linkedHashMap = new LinkedHashMap<>();
           JSONArray val = values.getJSONArray(j);
-          linkedHashMap.put(TIMESTAMP,
+          linkedHashMap.put(prometheusResponseFieldNames.getTimestampFieldName(),
               new ExprTimestampValue(Instant.ofEpochMilli((long) (val.getDouble(0) * 1000))));
-          linkedHashMap.put(VALUE, new ExprDoubleValue(val.getDouble(1)));
-          linkedHashMap.put(LABELS, new ExprStringValue(metric.toString()));
+          linkedHashMap.put(prometheusResponseFieldNames.getValueFieldName(), getValue(val, 1,
+              prometheusResponseFieldNames.getValueType()));
+          if (isQueryRangeFunctionScan) {
+            linkedHashMap.put(LABELS, new ExprStringValue(metric.toString()));
+          } else {
+            insertLabels(linkedHashMap, metric);
+          }
```

Comment:
Could you add some comments or UT? I don't quite understand what is happening here...

Reply:
UTs covered in this class.
https://github.com/opensearch-project/sql/blob/main/prometheus/src/test/java/org/opensearch/sql/prometheus/storage/PrometheusMetricScanTest.java#L43-L43

Added long description on the decision. Since query_range is out of scope for 2.4, will come up with other alternate solutions to handle this.